src/PlaneWaveBasis.jl

```diff
@@ -568,3 +568,35 @@ function gather_kpts(data::AbstractArray, basis::PlaneWaveBasis)
         nothing
     end
 end
+
+# Packing routines used in direct_minimization and newton algorithms.
+# They pack / unpack sets of ψ's (or compatible arrays, such as hamiltonian
+# applies and gradients) to make them compatible to be used in algorithms
+# from IterativeSolvers.
+# Some care is needed here : some operators (for instance K in newton.jl)
+# are real-linear but not complex-linear. To overcome this difficulty, instead of
+# seeing them as operators from C^N to C^N, we see them as
+# operators from R^2N to R^2N. In practice, this is done with the
+# reinterpret function from julia.
+# /!\ pack_ψ does not share memory while unpack_ψ does
+function pack_ψ(basis::PlaneWaveBasis{T}, ψ) where T
```

Comment:
This is a pretty short name given how internal it is and how specific the use case. Also I think the unusual behaviour to change the type from complex to real and back should be made explicit in the name, otherwise that is too magical.

Suggestions: `pack_densereal_ψ` and `unpack_densereal_ψ`?

Reply:
I know that makes the name quite long, but you bind the `basis` argument anyway whenever you use it, so that should be ok.

Reply:
I like what you suggest ! @antoine-levitt ok for you ?

Reply:
densereal is unreadable. pack_dense_real_psi is very long... The problem here is that this does two things 1) pack 2) realify. We should split this so we have imag_to_real(pack_psi(psi)), which makes pack_psi less of an issue. Agree? If yes @gkemlin you can just define imag_to_real and real_to_imag (working on flat arrays) to just do the reinterpret. Then pack works on a psi and produces a complex flat array, and unpack works on a complex flat array and produces a psi.

Reply:
Yup, ok, I can try to do that if it is good for both of you.

Reply:
I like the idea of splitting the functionality into two functions. `complex_to_real` is a bit vague. How about `reinterpret_real` and `reinterpret_complex` That are function names for which we could define more matching methods in different contexts if needed.

Reply:
yup that's good